api/http/api_deployments.go

```diff
@@ -394,124 +410,57 @@ func (d *DeploymentsApiHandlers) GenerateImage(w rest.ResponseWriter, r *rest.Re
-	bytes, err := ioutil.ReadAll(metaReader)
-	if err != nil && err != io.EOF {
-		return nil, errors.Wrap(err, "Failed to obtain value for "+p.FormName())
+	file, fileHeader, err := r.FormFile("artifact")
+	if err != nil {
+		return nil, errors.Wrap(err, "request does not contain the file")
+	}
+
+	if size < 0 || size > 0 && size != fileHeader.Size {
+		return nil, errors.New("The size value is wrong.")
 	}
 
-	strValue := string(bytes)
-	return &strValue, nil
+	multipartUploadMsg.ArtifactReader = file
+	multipartUploadMsg.ArtifactSize = fileHeader.Size
+
+	return multipartUploadMsg, nil
 }
 
 // ParseGenerateImageMultipart parses multipart/form-data message.
-func (d *DeploymentsApiHandlers) ParseGenerateImageMultipart(mr *multipart.Reader, maxMetaSize int64) (*model.MultipartGenerateImageMsg, error) {
+func (d *DeploymentsApiHandlers) ParseGenerateImageMultipart(r *rest.Request) (*model.MultipartGenerateImageMsg, error) {
 	multipartGenerateImageMsg := &model.MultipartGenerateImageMsg{}
-	for {
-		p, err := mr.NextPart()
-		if err != nil {
-			return nil, errors.Wrap(err, "Request does not contain artifact")
-		}
-		switch p.FormName() {
-		case "name":
-			name, err := d.getFormFieldValue(p, maxMetaSize)
-			if err != nil {
-				return nil, err
-			}
-			multipartGenerateImageMsg.Name = *name
-		case "description":
-			desc, err := d.getFormFieldValue(p, maxMetaSize)
-			if err != nil {
-				return nil, err
-			}
-			multipartGenerateImageMsg.Description = *desc
-		case "size":
-			size, err := d.getFormFieldValue(p, maxMetaSize)
-			if err != nil {
-				return nil, err
-			}
-			multipartGenerateImageMsg.Size, err = strconv.ParseInt(*size, 10, 64)
-			if err != nil {
-				return nil, err
-			}
-		case "device_types_compatible":
-			deviceTypesCompatible, err := d.getFormFieldValue(p, maxMetaSize)
-			if err != nil {
-				return nil, err
-			}
-			multipartGenerateImageMsg.DeviceTypesCompatible = strings.Split(*deviceTypesCompatible, ",")
-		case "type":
-			fileType, err := d.getFormFieldValue(p, maxMetaSize)
-			if err != nil {
-				return nil, err
-			}
-			multipartGenerateImageMsg.Type = *fileType
-		case "args":
-			args, err := d.getFormFieldValue(p, maxMetaSize)
-			if err != nil {
-				return nil, err
-			}
-			multipartGenerateImageMsg.Args = *args
-		case "file":
-			// file size part should be provided before artifact part
-			// file size value should be greater then 0
-			if multipartGenerateImageMsg.Size <= 0 {
-				return nil, errors.New("No size provided before the file part of the message or the size value is wrong.")
-			}
-			// HTML form can't set specific content-type, it's automatic, if not empty - it's a file
-			if p.Header.Get("Content-Type") == "" {
-				return nil, errors.New("The last part of the multipart/form-data message should be a file.")
-			}
-			multipartGenerateImageMsg.FileReader = p
-			return multipartGenerateImageMsg, nil
-		}
+	multipartGenerateImageMsg.Name = r.FormValue("name")
```

Comment:
what if there required field will be empty?
same with type
shouldn't device_type_compatible be required too?

Reply:
@kjaskiewiczz and type as well, I've added the checks